compiler/src/dmd/expressionsem.d

```diff
@@ -5024,7 +5024,28 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
 
             if (!exp.ignoreAttributes)
                 checkFunctionAttributes(exp, sc, exp.f);
-            checkAccess(exp.loc, sc, ue.e1, exp.f);
+
+            // Cut-down version of checkAccess() that doesn't use the "most visible" version of exp.f.
+            // We've already selected an overload here.
+            const parent = exp.f.toParent();
+            if (parent && parent.isTemplateInstance())
+            {
+                // already a deprecation
+            }
+            else if (exp.f.type.isTypeFunction().isref)
+            {
+                // already a deprecation
+            }
```

Comment:
How come we already issue deprecations for this case but not for the one you are adding now? It seems like this case should also be checked for along with the previous ones.

Reply:
21275 is "this bug, but ran into on the path to resolve overloads during an assignment that's lowered to a function call". I'll check tomorrow if I can just cover 21275 with my fix.

Reply:
Case removed.